### ISSUE TITLE
New Rule: space unary ops [$15 awarded]

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#1346]
This rule should require or disallow spaces before or after unary operations. For example:

``` js
var foo = ! b;    // disallow this
var bar = c ++;  // disallow this
var baz = void foo; // enforce this
```

The rule ID should be `space-unary-ops` and have options for "always" or "never". This rule would supercede and replace `space-unary-word-ops`.
##

**Comment by t128** [comment]
```javascript
this.myNum++;
```

throws a *space-unary-ops* error

**Comment by michaelficarra** [comment]
@t128 That has been fixed by #2764.

**Comment by t128** [comment]
thank you, found the fix after posting this comment
